src/main/java/lotto/enums/LottoRank.java

```diff
@@ -6,28 +6,50 @@
 
 public enum LottoRank {
     FIRST(6, 2_000_000_000),
-    SECOND(5, 1_500_000),
-    THIRD(4, 50_000),
-    FOURTH(3, 5_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
     LOSE(-1,0);
 
     private int matchedCount;
     private int price;
 
+    public static final int RANK_SECOND_MATCHED_COUNT = 5;
+
     LottoRank(int matchedCount, int price) {
         this.matchedCount = matchedCount;
         this.price = price;
     }
 
-    public static LottoRank matchRank(int matchedCount) {
+    public static LottoRank valueOf(int matchedCount, boolean bonusMatched) {
+        if (bonusMatched) {
+            return rankWithBonus(matchedCount);
+        }
+        return rankWithoutBonus(matchedCount);
+    }
+
+    private static LottoRank rankWithBonus(int matchedCount) {
+        if (matchedCount == RANK_SECOND_MATCHED_COUNT) {
+            return SECOND;
+        }
+        return rankWithoutBonus(matchedCount);
+    }
+
+    public static LottoRank rankWithoutBonus(int matchedCount) {
         return Arrays.stream(LottoRank.values())
                 .filter(r -> r.getMatchedCount() == matchedCount)
+                .filter(r -> !r.equals(SECOND))
```

Comment:
요청사항 구현 깔끔하게 잘해주셨습니다 💯 
소소한 의견 드리자면 `filter`에 해당 조건을 넣는 방법도 있습니다

```suggestion
                .filter(r -> !r.equals(SECOND) || bonusMatched)
```
근데 수연님이 작성하신게 가독성 측면에선 훨씬 좋네요 (이해하기 쉽고) 😅 
이 부분도 참고만 해주시면 감사하겠습니다 (굳이 수정 안하셔도 됩니다) 🙏